okhttp-client-handler/monix/src/main/scala/com/softwaremill/sttp/okhttp/monix/OkHttpMonixClientHandler.scala

```diff
@@ -0,0 +1,70 @@
+package com.softwaremill.sttp.okhttp.monix
+
+import java.nio.ByteBuffer
+
+import com.softwaremill.sttp._
+import com.softwaremill.sttp.okhttp.OkHttpAsyncClientHandler
+import monix.eval.Task
+import monix.execution.{Cancelable, Scheduler}
+import monix.reactive.{Consumer, Observable}
+import okhttp3.{MediaType, OkHttpClient, RequestBody => OkHttpRequestBody}
+import okio.BufferedSink
+
+import scala.language.higherKinds
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Created by omainegra on 8/4/17.
+  */
+class OkHttpMonixClientHandler private (client: OkHttpClient)(
+    implicit s: Scheduler)
+    extends OkHttpAsyncClientHandler[Task, Observable[ByteBuffer]](client,
+                                                                   TaskMonad) {
+
+  override def streamToRequestBody(
+      stream: Observable[ByteBuffer]): Option[OkHttpRequestBody] =
+    Some(new OkHttpRequestBody() {
+      override def writeTo(sink: BufferedSink): Unit =
+        stream
+          .consumeWith(
+            Consumer.foreach(chunck => sink.write(chunck.array()))
+          )
+          .runAsync
+
+      override def contentType(): MediaType = null
+    })
+
+  override def responseBodyToStream(
+      res: okhttp3.Response): Try[Observable[ByteBuffer]] =
+    Success(
+      Observable.fromInputStream(res.body().byteStream()).map(ByteBuffer.wrap))
```

Comment:
I think everything that uses an `InputStream` is potentially blocking - as the `InputStream` interface has no way of signalling "waiting for more input, check back later" or "give me a callback to signal you when new data is ready".

Looking at the implementation of `RealBufferedSource.inputStream()`, it looks like data is buffered into memory, and if it is not available, a potentially blocking call to `source.read` is done. So this is definitely useful as well, but again, I think it would be worth clearly documenting that the call might block a thread - I suppose in this case it could be on the the `Scheduler`'s threads?